ALFA audio analogue synthesizer AS33xx ICs

robkam opened the pull request:
Adding ALFA 3310, 3320, 3330, 3340, 3345 and 3360 and Cool Audio V3320 and V3340 audio analogue synthesizer ICs. New parts compatible with those (similarly named) previously from Curtis Electromusic Specialities (CEM), now known as OnChip Systems.

http://www.alfarzpp.lv/eng/sc/AS3310.pdf
http://www.alfarzpp.lv/eng/sc/AS3320.pdf
http://www.alfarzpp.lv/eng/sc/AS3330.pdf
http://www.alfarzpp.lv/eng/sc/AS3340.pdf
http://www.alfarzpp.lv/eng/sc/AS3360.pdf
[Cool Audio_Data_Sheet_V3320_VCF.pdf](https://github.com/KiCad/kicad-symbols/files/1818601/Cool.Audio_Data_Sheet_V3320_VCF.pdf)
[COOLAUDIO_V3340_DATASHEETS.pdf](https://github.com/KiCad/kicad-symbols/files/1818603/COOLAUDIO_V3340_DATASHEETS.pdf)

Misca1234 commented:
You might disagree to the following but

CEM3310
https://static.miraheze.org/sdiywiki/f/ff/CEM3310_datasheet_long.pdf
I get the pins to this
1 out
2 out
3 out
Figure 1 and 3 on page 4

You should stack pin 6 and 7
8 passive (usually decoupling pins are made passive)

is there a special reason why you choose 
Package_DIP:DIP-16_W7.62mm_Socket_LongPads
and not the "normal" one, like
DIP-16_W7.62mm
-------------------------
CEM3320
https://static.miraheze.org/sdiywiki/d/d1/CEM3320_datasheet_long.pdf
4,11,16 passive (usually decoupling pins are made passive)

Same question goes for your choice of foot print
-------------------------
CEM3330/3335
https://static.miraheze.org/sdiywiki/5/5c/CEM_3330-3335_long.pdf

pin 1 out   (Figure 1 on page 1)
9, 10 passive   (usually decoupling pins are made passive)

Same question goes for your choice of foot print
-------------------------
CEM3340
https://static.miraheze.org/sdiywiki/e/e4/CEM_3340-3345_long.pdf
11 passive   (usually decoupling pins are made passive)
14 output   (Figure 1 on page 1)
I guess the precision multiplier generate a voltage over the external resistor, which mean the 
voltage is coming form within.
Same question goes for your choice of foot print
-----------

You forgot to add CEM3335, CEM3345

poeschlr commented:
I would also ask why using a socket footprint by default. Use the normal DIP footprint and write the footprint filter such that the sockets can be selected as an alternative.

robkam commented:
@poeschlr the ICs are usually socket mounted. I'd have preferred a socket with IC in place, but couldn't find it. I forget now, why I chose long pads.

Please could you link to an example of a filtered footprint?

poeschlr commented:
What is your source for the claim that these are usually socket mounted?

robkam commented:
examples 

http://www.doepfer.de/A100_pictures/A-111-2.jpg commercial product with 3340 socketed 

http://www.digisound80.co.uk/digisound/modules/80-2_files/80-2&3_ETI.pdf mentions testing socket voltages before inserting 3340

http://www.mindburner.co.uk/synth/prophet5_fault.html mentions swapping out faulty 3340s and being careful not to damage as they're expensive to replace

robkam commented:
I don't know of any CEM3345 outside of the data sheet and the CEM3335 only rarely.

robkam commented:
@evanshultz  thanks for your help. CEM (now OnChip) ceased producing and these sought after ICs became rare and expensive. Recently with the patents expiring Alfa in Latvia have started making them easily available again - how does this affect their library designations? There are new data sheets at http://www.alfarzpp.lv/eng/sc/application.php

poeschlr commented:
In that case please add the replacement part instead of the obsolete parts. (you can mention in the description that these are pin compatible to the obsolete part.)

Misca1234 commented:
Well, now when the footprint is up to discussion.
I suggest you set you FP filter to
DIP*W7.62mm*
And then you leave the "dedicated and specific" foot print filter empty.
In this way you are able to choose any "mounting type" you want when you run the CvPcb

> . Recently with the patents expiring Alfa in Latvia have started making them easily available again

That explains a lot, I was not going to say anything if it would kill your interest in contribute with a symbol, but I was really sceptical when I saw the data sheet. First it is dated 1979 and it have an image of the good old fashion cermaic DIP, I have not seen those since the 80's in the military industry and was thinking why would he want to add this chip :).

evanshultz commented:
@robkam 
With all the changes you've made, please post fresh screenshots of the symbols.

General:
1. Are you intending to only add these parts in DIP? They are all available in at least two packages.
1. It seems these datasheets aren't consistent with pin names regarding capitalization, using an underscore, smashing all the letters together, or using subscript. You can deviate from the below suggestions in this regard as long as you're consistent and following the datasheet as much as possible.
1. Along with the above, it seems like you may have used the CEM pin names or some colloquial pin names. Since you've added the ALFA datasheets, IMO we should use those pins names.
1. I would consider putting the original name in the description, such as `CEM3310`. I would think folks would have the original PN, with the manufacturer prefix and not just the numbers, in their mind.

CS3310:
1. Pin 1: Passive type.
1. Pin 2: Name is not `Env_Out` or `EnvOut` (or all caps in either form) from datasheet.
1. Pin 3: Name should be `Vp` and type should be Input.
1. Pin 7: Name should be `PGND`.
1. Pin 8: Name should be `CCOMP` or `C_COMP`.
1. Pin 9: Name should be `VCS` or `V_CS`.
1. Pin 12: Name should be `VCD` or `V_CD`.
1. Pin 13: Name should be `VCR` or `V_CR`.
1. Pin 14: Is there any chance this pin should be directly tied to pin 7? The datasheet doesn't mention they should be and in analog circuits like this sometimes a bit of ground impedance can be good. I think unstacking them may be best.
1. Pin 15: Name should be `VCA` or `V_CA`.
1. Pin 16: Name should be `ATK_OUT` or `ATK_OUT`.
1. It is missing "ADSR " at the beginning of the description.

AS3320:
1. You've used A and B instead of 1 and 2; for example, pins 1 and 2 should be `IN1` and `IN2`.
1. Pin 8: Name is `VRES`.
1. Pin 9: Name is `IRES`.
1. Pin 12: Name is `VCFI`.

AS3330:
1. Again, you've used A and B instead of 1 and 2; for example, pin 1 should be `OUT1`.
1. Pin 2: Name should be `VG1`.
1. Pin 6: Name is `VCE1`.
1. Pin 7: Name is `VCL1`.
1. Pin 8: I would make it Passive type since it's only a resistor a negative supply.
1. Pin 9: Name is `CCOMP1`.
1. The above goes for VCA2 as well.

AS3340:
1. Pin 1: Name is `SCALE1`.
1. Pin 2: Name is `SCALE2`.
1. Pin 3: Name is `VP`.
1. Pin 5: Name is `VPWM`.
1. Pin 6: Name is `VHSI`.
1. Pin 7: Name is `VHFT` and it looks to me like the type should be Input.
1. Pin 8: Name is `VSO`.
1. Pin 9: Name is `VSSI`.
1. Pin 10: Name is `VTO`.
1. Pin 11: Name is `CAP`.
1. Pin 13: Name is `VLFI`.
1. Pin 14: Seems like it should be type Passive.
1. Pin 15: Name is `VFCI`.

AS3360:
1. Pin 2: Name is `IO1`.
1. Pin 3: Name is `VE1`.
1. Pin 4: Name is `VO1`.
1. Pin 5: Name is `VC1`.
1. Pin 8: It looks like this is an internal reference voltage from which you need cannot pull any current but it needs filtering. In that case, Passive type is better.
1. As above, I won't repeat the comments for VCA2.

robkam commented:
@evanshultz Thanks for the help.
I'm changing pin names to ALFA.
I'm intending to add the other packages, (SOIC & QFN).
With the AS3330 and AS3360 I'll separate the unit parts.

robkam commented:
3345 some pins still wrong

evanshultz commented:
@robkam 
- The commit descriptions are sometimes poor. Can you try to make sure that the commit description actually matches what was done? For example, `Update Audio.lib` isn't descriptive. It's especially hard to keep track of things when there are so many commits.
- It seems you've added an ALIAS for the CEM parts. I thought only the ALFA parts were going to added since the CEM parts were discontinued.

Let me take a closer look now...

robkam commented:
@evanshultz Will do better descriptions. Recent changes were minor, to fix errors or warnings in the build log.

OnChip recently produced a fresh batch of CEM3340s and although dormant haven't officially discontinued any.

robkam commented:
Changes to multi unit for AS3330 and AS3360 got lost - will do them again. Then to add other packages, SOIC, QFN & SOP.

evanshultz commented:
@robkam 
Thank you for making more detailed commit messages.

Here's how I want to proceed:
1. Get the AS3360 back.
1. Let me review the commit without further changes from you.
1. Resolve any issue there and then I merge those symbols.
1. Add other packages in a later PR.

Let me know if you agree. Then advise when you get the lost symbol back.

robkam commented:
@evanshultz  
1. AS3330 and AS3360 are back (CEM component/alias has the AS name?)
2. ok I'll leave alone until next reviewed
3. a merge would be great
4. okay to add other packages later

I'm not yet certain that the pins are positioned so as to help keep a schematic tidy.

Thanks for your help and patience.

robkam commented:
CoolAudio V3320  is SOP-18L, need to rm alias.

evanshultz commented:
@robkam 
I requested new screenshots of the symbols above. This helps review whenever there's a visual change.
![image](https://user-images.githubusercontent.com/1936989/37547109-fcc65e3a-292c-11e8-9190-9054d501b37d.png)

General:
- Would you mind just sticking with the ALFA parts? The others seem to be sporadic or old, from what you're writing. Including them more than doubles the numbers of apparent symbols in the library for little to no benefit, IMO.
- The datasheet link needs to be a single one. The CEM parts have two with a slash in between and that doesn't work.
- Mixing some lowercase and some mixed case pin names looks unpleasant to me. Perhaps always use mixed case? AS3320 is particularly bad; perhaps we ignore the datasheet in this case for consistency?
- Set the pin text offset to 20mil. This will help with overlapping or nearly-overlapping text.
- Place the ref des to the left of the power pin on top and the MPn to the right of that pin.
- In general these symbols are rather cramped. There will be a lot of passive parts around them. Perhaps enlarge them a bit and spread out the pins. AS3320, for example, has 4 sections but their pins are all smashed together on the left side.

AS3310:
- Put `ASDR ` at the front of the description, not as a keyword. This matches the datasheet.
- Should pins 7 and 14 always be tied together?
- Overlapping pin names need to be fixed.
- Maybe move the output pins down to the middle of the right side and separate them?

AS3330:
- What are the Dtrim pins used for? What connects to them?
- Maybe move the output pin down to the middle of the right side?

AS3340:
- Maybe move the output pins down to the middle of the right side and separate them?

AS3345:
- Can pins 13 and 14 be stacked?
- Pin 7 should be Output type.
- Maybe move the output pins down to the middle of the right side and separate them?

AS3360:
- Pin 8 looks like this is an internal reference voltage from which you need cannot pull any current but which needs filtering. In that case, Passive type is better.
- There are two pin 8s.
- Maybe move the output pins down to the middle of the right side and separate them?

robkam commented:
AS3330 pin 3 and 17 if it's required to reduce distortion go to a potential divider between VCC/VEE and GND otherwise just to GND.
![as3330 pin 3 or 17](https://user-images.githubusercontent.com/4047844/37554727-ba4c136c-29d4-11e8-999b-ec9527980b5b.PNG)

robkam commented:
![as33xx 17-03-2018](https://user-images.githubusercontent.com/4047844/37556217-ab35cd76-29ea-11e8-937c-6b88df53258e.png)
Most suggestions now done, except: 
I've left AS3330 DTRIM pins 3 and 17 as input (see previous comment).
Uncramping and spreading the pins out more, to suit neater schematic.

Deleted all CEM aliases, kept as keywords.
Names now all upper case.
Text offset now 20mil.
Ref des’s & MPns moved.
Moved and spread o/p pins.
AS3310 - renamed pin 7 to GND, stacked pins 7 & 14. Power and signal ground are the same here.
AS3345 - 13 and 14 stacked
AS3340 - pin 7 now o/p
AS3360 - pin 8 fixed

evanshultz commented:
These symbols are looking good now! Thanks for taking time to polish them up.
- In the keywords for AS3310 you have `Alfa`, with mixed case, but all others are `ALFA`. Can you standardize?
- AS3330 DTRIM pins should be Passive type, then, because only resistors are connected. This is most strongly driven by the ERC engine in KiCad.
- Similarly, this is why AS3340 pin 7 was Passive type. Also AS3345 pin 8.
- Are the AS3310 ground pins *always* connected together? If there's even the possibility that they may not be, like if a few ohm should be between them or even a 0ohm resistor as a jumper, those pins should not be stacked. Unless there is documentation saying they must always be connected together, do not stack.
- AS3360 pin 8 should not be Input type. It's an internal reference which, as far as I can tell, can not source current. So Passive type is good since only a cap is connected.

robkam commented:
Can Travis CI log be ignored at the following?
New symbol 'Audio.lib:AS3360'
Checking symbol 'AS3360':
  Violating S4.4
    Pin electrical type should match pin function
    Pin types should match pin function
     - Pin VREF (8) @ (-400,-100) is type PASSIVE : suggested INPUT

robkam commented:
![capture](https://user-images.githubusercontent.com/4047844/37567919-d5daccbc-2ac5-11e8-94b2-868fdb2b2059.PNG)

evanshultz commented:
@robkam 
Yes, this warning can be ignored. It does not apply here. Unfortunately our scripts still have a few false positives.

Only a couple more things:
- AS3330: Both pins 3 and 17 have the same name.
- AS334x: The VHFT pin type changed from Input.

evanshultz commented:
You changed VHFT on AS3340 and but not AS3345.

Once that's done, are you ready for these symbols to be merged?

evanshultz commented:
I'll take it you are ready to merge then. Thanks for contributing these symbols and sticking with it to get them finished.

robkam commented:
Thanks!
I'm glad to have been able to make a tiny contribution to KiCad :-D.
